.buildkite/check-trailing-whitespace.sh

```diff
@@ -4,7 +4,7 @@
 # SPDX-License-Identifier: LicenseRef-MIT-TQ
 
 files=$(git ls-files -- . ':!:*.patch' | xargs grep --files-with-matches --binary-files=without-match '[[:blank:]]$')
-if [[ ! -z $files ]];then
+if [[ -n $files ]];then
```

Comment:
According to [SC2236](https://www.shellcheck.net/wiki/SC2236), `-n` absolutely requires quoting variables.

```suggestion
if [[ -n "$files" ]];then
```